src/main/java/seedu/hirehub/model/application/UniqueApplicationList.java

```diff
@@ -0,0 +1,154 @@
+package seedu.hirehub.model.application;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.hirehub.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.hirehub.model.application.exceptions.ApplicationNotFoundException;
+import seedu.hirehub.model.application.exceptions.DuplicateApplicationException;
+
+/**
+ * A list of applications that enforces uniqueness between its elements and does not allow nulls.
+ * An application is considered unique by comparing using {@code Application#isSameApplication(Application)}.
+ * As such, adding and updating of applications uses Application#isSameApplication(Application) for equality
+ * so as to ensure that the application being added or updated is
+ * unique in terms of identity in the UniqueApplicationList. However, the removal of a job uses
+ * Application#equals(Object) so as to ensure that the application with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Application#isSameApplication(Application)
+ */
+public class UniqueApplicationList implements Iterable<Application> {
+
+    private final ObservableList<Application> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Application> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent application as the given argument.
+     */
+    public boolean containsApplication(Application toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameApplication);
+    }
+
+    /**
+     * Adds an application to the list.
+     * The application must not already exist in the list.
+     */
+    public void addApplication(Application toAdd) {
+        requireNonNull(toAdd);
+        if (containsApplication(toAdd)) {
+            throw new DuplicateApplicationException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the application {@code target} in the list with {@code editedApplication}.
+     * {@code target} must exist in the list.
+     * The job identity of {@code editedApplication} must not be the same as another existing job in the list.
+     */
+    public void setJob(Application target, Application editedApplication) {
```

Comment:
Should the method name be setApplication?

Reply:
I think some got merged when resolving merge conflict; lemme fix also